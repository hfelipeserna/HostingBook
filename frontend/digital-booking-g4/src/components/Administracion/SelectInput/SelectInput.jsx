import styles from "./SelectInput.module.css"

export default function SelectInput({
    label,
    name,
    handleChange,
    opcionesDisponibles,
    optionDefault,
    showOptions
}) {
    const horaFormat = (hora) => {
        if (hora > 12) {
            const h = hora - 12;
            return ((h < 10) ? ("0" + h + ":00 PM") : (h + ":00 PM"));
        } else {
            return ((hora < 10) ? ("0" + hora + ":00 AM") : (hora + ":00 AM"));
        }
    }


    return (
        <div className={styles.containerSelect}>
            <label>{label}</label>
            <select name={name} className={styles.minimal} onChange={handleChange} required>
                <option value={name} className={styles.optionDefault} hidden>{optionDefault}</option>
                {opcionesDisponibles.map((o, i) => (
                    <option value={o} key={`opcionesDisponibles-${i}`}>{showOptions? horaFormat(o) : o}</option>
                ))}
            </select>
        </div>
    )
}