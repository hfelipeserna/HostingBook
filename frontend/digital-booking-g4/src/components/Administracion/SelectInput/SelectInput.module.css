.containerSelect{
    display: flex;
    flex-direction: column;
    width: 100%;
    position: relative;
    
}

.containerSelect label{
    color: var(--color-2);
    font-weight: 700;
    margin-top: 20px;
    cursor: pointer;
}

.containerSelect span{
    color: #BEBEBE;
    font-size: 0.75rem;
    font-weight: 500;
    position: absolute;
    left: 15px;
    top: 57px;

}

.containerSelect select{
    /* reset */
    margin: 0;      
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
    -webkit-appearance: none;
    -moz-appearance: none;

} 

option {
    font-size: 1rem ;
    color: var(--color-3);
    font-weight: 700;

}

.optionDefault{
    color: #BEBEBE;
    font-size: 0.75rem;
    font-weight: 500;
}

select.minimal {
    /* styling */
    background-color: var(--color-5);
    border: 3px solid var(--color-5);
    box-shadow: 0px 1px 5px rgb(0 0 0 / 15%);
    color: var(--color-2);
    font-size: 1rem;
    padding: 1vh 40px 1vh 15px;
    width: 100%;
    margin-top: 5px;

    /* Cambiar la flecha del select solo con css*/
    background-image:
      linear-gradient(45deg, transparent 50%, var(--color-principal) 50%),
      linear-gradient(135deg, var(--color-principal) 50%, transparent 50%),
      linear-gradient(to right, var(--color-principal), var(--color-principal));
    background-position:
      calc(100% - 20px) calc(1em + 2px),
      calc(100% - 15px) calc(1em + 2px),
      calc(100% - 2.5em) 0.5em;
    background-size:
      5px 5px,
      5px 5px,
      1px 1.5em;
    background-repeat: no-repeat;
}
  
select.minimal:focus {
    background-image:
      linear-gradient(45deg, var(--color-principal) 50%, transparent 50%),
      linear-gradient(135deg, transparent 50%, var(--color-principal) 50%),
      linear-gradient(to right, var(--color-principal), var(--color-principal));
    background-position:
      calc(100% - 15px) 1em,
      calc(100% - 20px) 1em,
      calc(100% - 2.5em) 0.5em;
    background-size:
      5px 5px,
      5px 5px,
      1px 1.5em;
    background-repeat: no-repeat;
    border-color: var(--color-principal);
    outline: 0;
}


  
  
select:-moz-focusring {
    color: transparent;
    text-shadow: 0 0 0 #000;
}

@media (max-width: 480px) {
    .containerSelect select{
        /* styling */
        width: 100%;
    }

    .containerSelect label{
        margin-top: 7px;
    }
}