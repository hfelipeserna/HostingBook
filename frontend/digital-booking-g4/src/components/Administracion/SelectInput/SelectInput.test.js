import SelectInput from "./SelectInput"
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react"
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import { configure } from "enzyme";

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Select Input test", function () {
    const label = "este es el label"
    const nombre = "este es el nombre"
    const handleChange = () => {

    }
    const opcionesDisponibles = ["10", "9", "21", "13"]
    const opcionDefault = "esta es la opcionDefault"
    let showOptions = "true"


    test("Renderiza Componente Con show opcions true", () => {
        const renderizacion = render(<SelectInput opcionesDisponibles={opcionesDisponibles} optionDefault={opcionDefault} showOptions={showOptions} handleChange={handleChange}></SelectInput>)
        expect(renderizacion).toMatchSnapshot()

    });
    test("Renderiza Componente Con show opcions false", () => {
        showOptions = false
        const renderizacion = render(<SelectInput opcionesDisponibles={opcionesDisponibles} optionDefault={opcionDefault} showOptions={showOptions} handleChange={handleChange}></SelectInput>)
        expect(renderizacion).toMatchSnapshot()

    });


});
