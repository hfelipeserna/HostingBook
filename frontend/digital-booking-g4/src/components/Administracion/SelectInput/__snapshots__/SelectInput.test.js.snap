// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Input test Renderiza Componente Con show opcions false 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="containerSelect"
      >
        <label />
        <select
          class="minimal"
          required=""
        >
          <option
            class="optionDefault"
            hidden=""
          >
            esta es la opcionDefault
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="9"
          >
            9
          </option>
          <option
            value="21"
          >
            21
          </option>
          <option
            value="13"
          >
            13
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="containerSelect"
    >
      <label />
      <select
        class="minimal"
        required=""
      >
        <option
          class="optionDefault"
          hidden=""
        >
          esta es la opcionDefault
        </option>
        <option
          value="10"
        >
          10
        </option>
        <option
          value="9"
        >
          9
        </option>
        <option
          value="21"
        >
          21
        </option>
        <option
          value="13"
        >
          13
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Select Input test Renderiza Componente Con show opcions true 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="containerSelect"
      >
        <label />
        <select
          class="minimal"
          required=""
        >
          <option
            class="optionDefault"
            hidden=""
          >
            esta es la opcionDefault
          </option>
          <option
            value="10"
          >
            10:00 AM
          </option>
          <option
            value="9"
          >
            09:00 AM
          </option>
          <option
            value="21"
          >
            09:00 PM
          </option>
          <option
            value="13"
          >
            01:00 PM
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="containerSelect"
    >
      <label />
      <select
        class="minimal"
        required=""
      >
        <option
          class="optionDefault"
          hidden=""
        >
          esta es la opcionDefault
        </option>
        <option
          value="10"
        >
          10:00 AM
        </option>
        <option
          value="9"
        >
          09:00 AM
        </option>
        <option
          value="21"
        >
          09:00 PM
        </option>
        <option
          value="13"
        >
          01:00 PM
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
